runtime/near-vm-errors/src/lib.rs

```diff
@@ -174,7 +174,7 @@ impl std::fmt::Display for HostError {
             GasLimitExceeded => write!(f, "Exceeded the maximum amount of gas allowed to burn per contract."),
             BalanceExceeded => write!(f, "Exceeded the account balance."),
             EmptyMethodName => write!(f, "Tried to call an empty method name."),
-            GuestPanic => write!(f, "Smart contract has explicitly invoked `panic`."),
+            GuestPanic(s) => write!(f, "Smart contract panicked: {}", s),
```

Comment:
should we escape the msg?

Reply:
No. I think we should refactor FunctionCallError to be an enum with typed errors. And panic with a message should be one of them .